File: exercises/depth_of_tree_104.java
import java.lang.Math;
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode() {}
    TreeNode(int val) { this.val = val; }
    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }
}

class Solution {
    public int maxDepth(TreeNode root) {
        return depth(root, 0);
    }
    int depth (TreeNode node, int height) {
        if (node == null) {
            return height;
        }
        return Math.max(depth(node.left, height+1), depth(node.right, height+1));
    }
}

File: exercises/depth_of_tree_104.py
from typing import Optional
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        return self.depth(root, 0)
        
    def depth(self, node: Optional[TreeNode], height: int) -> int:
        if node is None:
            return height
        return max(self.depth(node.left, height+1), self.depth(node.right, height+1))

